### PR TITLE
Workload: low hanging fruit optimization: don't unneccessarily update() entities

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8609/app_12827_PR.apk

File: libraries/entities/src/PolyLineEntityItem.h [hunk 1]
```diff
@@ -78,8 +78,6 @@ class PolyLineEntityItem : public EntityItem {
     QString getTextures() const;
     void setTextures(const QString& textures);
 
-    virtual bool needsToCallUpdate() const override { return true; }
-
```
Comment (AndrewMeadows) [inline libraries/entities/src/PolyLineEntityItem.h]: This looks like a bug:  `PolyLineEntityItem` was reporting `true` for `needsToCallUpdate()` but didn't actually override the `update()` method, which means it gets the base class's implementation which just updates `_lastUpdate` timestamp.  However, I couldn't see find any code that actually _uses_ the line's `_lastUpdate` member.  But I'm not 100% sure I didn't miss something.  We'll have to test it.

File: libraries/entities-renderer/src/RenderableZoneEntityItem.h [hunk 2]
```diff
@@ -69,7 +69,6 @@ class ZoneEntityRenderer : public TypedEntityRenderer<ZoneEntityItem> {
     graphics::SkyboxPointer editSkybox() { return editBackground()->getSkybox(); }
     graphics::HazePointer editHaze() { _needHazeUpdate = true; return _haze; }
 
-    bool _needsInitialSimulation{ true };
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableZoneEntityItem.h]: This data member was unused by `RenderableZoneEntityItem`.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8610/app_12827_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8613/app_12827_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8610/HighFidelity-Beta-PR12827-353afb47b3a3f547234b6cd36d5ffe854b3d10bf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8613/HighFidelity-Beta-PR12827-9c84f0527527859251834e78f77f8bd1d61788fe.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8613/HighFidelity-Beta-PR12827-9c84f0527527859251834e78f77f8bd1d61788fe.exe

**Pull request merged**